### ISSUE TITLE
Add support for Maven 3.3.1 (required for polyglot projects) (reissued)

### ISSUE THREAD

**Issue by mariuszs** [travis-ci/travis-ci#4872]
This releated is closed and never working issue #3597, polyglot maven is still not working on travis.

You can verify this here: https://travis-ci.org/Codearte/catch-exception/builds/57842570

**Comment by BanzaiMan** [comment]
Sorry for the delay in build environment updates. Please override the `install` phase in the meantime.

**Comment by BanzaiMan** [comment]
Hmm. Maybe the version can't handle the `.groovy` file at all? Then I'm afraid the only workaround I can suggest is upgrading it yourself.

**Comment by mariuszs** [comment]
No. I have checked this and the maven version is `3.2.5`, too old for `pom.groovy`. Please take look again:

https://travis-ci.org/mariuszs/travis-maven-polyglot-test

**Comment by mariuszs** [comment]
```
Using worker: worker-linux-docker-450c5bac.prod.travis-ci.org:travis-linux-5
mvn -version
Apache Maven 3.2.5 (12a6b3acb947671f09b81f49094c53f426d8cea1; 2014-12-14T17:29:23+00:00)
Maven home: /usr/local/maven
Java version: 1.7.0_76, vendor: Oracle Corporation
Java home: /usr/lib/jvm/java-7-oracle/jre
Default locale: en_US, platform encoding: ANSI_X3.4-1968
OS name: "linux", version: "3.13.0-29-generic", arch: "amd64", family: "unix"
```

**Comment by mariuszs** [comment]
More information: [Maven Polyglot](https://github.com/takari/polyglot-maven)

> Polyglot for Maven is a set of extensions for Maven **3.3.1+** that allows the POM model to be written in dialects other than XML. Several of the dialects also allow inlined plugins: the Ruby, Groovy and Scala dialects allow this.

**Comment by 0xbaadf00d** [comment]
Hi @BanzaiMan,

When the version 3.3.X will be added ? Unfortunaly 3.2.5 version is still in use on Travis

This too old version dont handle correctly multiple modules pom.xml

**Comment by buckett** [comment]
As a workaround you could consider using: https://github.com/takari/maven-wrapper

But it would be really nice if travis would update to a newer maven version.

**Comment by kevinconaway** [comment]
Hi @BanzaiMan, any update on when maven 3.3+ will be added to the Trusty environment?  The current version is 3.1.1

> Sorry for the delay in build environment updates. Please override the install phase in the meantime.

Will you elaborate here? Is it possible to use the install phase to specify a different maven version?

**Comment by mariuszs** [comment]
Please use maven wrapper - this is best approach right now.

https://github.com/takari/maven-wrapper

**Comment by 0xbaadf00d** [comment]
maven wrapper is a workaround not a solution. @BanzaiMan  Is too difficult to upgrade Maven to prior version ?

**Comment by ctron** [comment]
Nearly one year and no progress?

**Comment by benhardy** [comment]
Could really use Maven 3.3.1. It's been out for a while now... c'mon guys.

**Comment by gallandarakhneorg** [comment]
+1 for Maven 3.3.9

**Comment by Phillipus** [comment]
+1 for Maven 3.3.x

**Comment by 0xbaadf00d** [comment]
I think this project is dead or the trainee in charge don't work any more on this project xD

**Comment by pavolloffay** [comment]
+1

**Comment by jschneider** [comment]
Ok, we have 2017 now. Any signs for support for Maven 3.3.9?

**Comment by rainleon** [comment]
+1 3.3.9

**Comment by javierjeronimo** [comment]
Found a workaround that works for me: https://gist.github.com/ppatierno/a50396b19c83554fe5f43a71c12c70c6
